crates/languages/src/rust/runnables.scm

```diff
@@ -25,3 +25,15 @@
     )
     (#set! tag rust-test)
 )
+
+; Rust main function
+(
+    (
+        (function_item
+            name: (_) @run
+            body: _
+        ) @end
```

Comment:
```suggestion
        ) @_rust_main_function_end
```

To continue thought from https://github.com/zed-industries/zed/pull/13087/files#r1644234771,
we could give this capture a distinct name and check whether it's set in `TaskContext` passed into `build_context`; if it is, that means that there's a main function in the current file. If not, we can avoid running `cargo metadata` altogether.

Reply:
@osiewicz  Thank you for the suggestion, I've updated the implementation as follows:
1. If it's a Rust main function, the package ID and bin target name are retrieved from `cargo metadata`.
2. If it's not a Rust main function, the package ID is retrieved from `cargo pkgid`.